libraries/lib-snapping/SnapUtils.h

```diff
@@ -0,0 +1,124 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  SnapUtils.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#pragma once
+
+#include "Prefs.h"
+#include "Registry.h"
+
+#include <functional>
+#include <string_view>
+
+enum class SnapMode
+{
+   SNAP_OFF,
+   SNAP_NEAREST,
+   SNAP_PRIOR
+};
+
+extern SNAPPING_API StringSetting SnapToSetting;
+extern SNAPPING_API EnumSetting<SnapMode> SnapModeSetting;
+
+SNAPPING_API SnapMode ReadSnapMode();
+SNAPPING_API Identifier ReadSnapTo();
+
+struct SnapRegistryGroup;
+struct SnapRegistryItem;
+
+struct SNAPPING_API SnapRegistryVisitor /* not final */
+{
+   virtual void BeginGroup(const SnapRegistryGroup& item) = 0;
+   virtual void EndGroup(const SnapRegistryGroup& item) = 0;
+   virtual void Visit(const SnapRegistryItem& item) = 0;
+};
+
+struct SNAPPING_API SnapConfig final
+{
+   double rate {};
+   double tempo {};
+   std::pair<int, int> timeSignature {};
+};
+
+struct SNAPPING_API SnapResult final
+{
+   double time {};
+   bool snapped {};
```

Comment:
Please comment what the second member means.

I had thought the meaning of the second member would be that the output time is different from the input time.  But that is not consistent with the examples of snap functions in SnapUtils.cpp.

If the purpose of the boolean is to indicate that a snap function is not found, while a snap function is expected always to return true in second place -- then a snap function should just return double.

If that is not the intent, please comment what it is, and change the snap functions so that they sometimes return false.